### ISSUE TITLE
[BUG] Do not parse interrupted request

### ISSUE THREAD

**Issue by akozich** [softwaremill/tapir#4169]
Tapir version: 1.11.7

Scala version: 3.5.0

Our endpoint receives the PUT request with JSON body from mobile clients. As mobile network may be unstable some requests are interrupted before receiving the full request body. 

We set `NettyConfig.requestTimeout` to 5s to make sure we close the connection from the server side. Then channel is marked inactive. When `org.playframework.netty.HandlerPublisher` receives `channelInactive` it completes by calling `SimpleSubscriber.onComplete` which sends all received bytes. As a result corrupted JSON is passed to the Codec.

What is the problem?

I'd propose if the number of bytes received is lower than the expected content length do not pass the bytes to the codec

**Comment by adamw** [comment]
Yeah this sounds like a bug, I suppose it should call `.onError` with a timeout exception?

Would you be able to create a PR maybe?

**Comment by sergiuszkierat** [comment]
@akozich I've prepared a draft [#4214](https://github.com/softwaremill/tapir/pull/4214) where I've tried to reproduce the issue. Could you please confirm if that test `NettyCatsRequestTimeoutTest` shows your scenario?

**Comment by rwalerow** [comment]
I did prepare a draft [#5466](https://github.com/softwaremill/tapir/pull/5466) to check 400 Bad Request response
